tests/test_updater_with_simulator.py

```diff
@@ -1,217 +0,0 @@
-
-    # set dump_dir to trigger repository state dumps
-    dump_dir: Optional[str] = None
-
-    def setUp(self) -> None:
-        # pylint: disable-next=consider-using-with
-        self.temp_dir = tempfile.TemporaryDirectory()
-        self.metadata_dir = os.path.join(self.temp_dir.name, "metadata")
-        self.targets_dir = os.path.join(self.temp_dir.name, "targets")
-        os.mkdir(self.metadata_dir)
-        os.mkdir(self.targets_dir)
-
-        # Setup the repository, bootstrap client root.json
-        self.sim = RepositorySimulator()
-        with open(os.path.join(self.metadata_dir, "root.json"), "bw") as f:
-            f.write(self.sim.signed_roots[0])
-
-        if self.dump_dir is not None:
-            # create test specific dump directory
-            name = self.id().split(".")[-1]
-            self.sim.dump_dir = os.path.join(self.dump_dir, name)
-            os.mkdir(self.sim.dump_dir)
-
-    def tearDown(self) -> None:
-        self.temp_dir.cleanup()
-
-    def _run_refresh(self) -> Updater:
-        """Creates a new updater and runs refresh."""
-        if self.sim.dump_dir is not None:
-            self.sim.write()
-
-        updater = Updater(
-            self.metadata_dir,
-            "https://example.com/metadata/",
-            self.targets_dir,
-            "https://example.com/targets/",
-            self.sim,
-        )
-        updater.refresh()
-        return updater
-
-    def test_refresh(self) -> None:
-        # Update top level metadata
-        self._run_refresh()
-
-        # New root (root needs to be explicitly signed)
-        self.sim.root.version += 1
-        self.sim.publish_root()
-
-        self._run_refresh()
-
-        # New timestamp
-        self.sim.update_timestamp()
-
-        self._run_refresh()
-
-        # New targets, snapshot, timestamp version
-        self.sim.targets.version += 1
-        self.sim.update_snapshot()
-
-        self._run_refresh()
-
-    def test_fishy_rolenames(self) -> None:
-        roles_to_filenames = {
-            "../a": "..%2Fa.json",
-            ".": "..json",
-            "/": "%2F.json",
-            "ö": "%C3%B6.json",
-        }
-
-        # Add new delegated targets, update the snapshot
-        spec_version = ".".join(SPECIFICATION_VERSION)
-        for rolename in roles_to_filenames:
-            role = DelegatedRole(rolename, [], 1, False, ["*"], None)
-            targets = Targets(1, spec_version, self.sim.safe_expiry, {}, None)
-            self.sim.add_delegation("targets", role, targets)
-        self.sim.update_snapshot()
-
-        updater = self._run_refresh()
-
-        # trigger updater to fetch the delegated metadata, check filenames
-        updater.get_targetinfo("anything")
-        local_metadata = os.listdir(self.metadata_dir)
-        for fname in roles_to_filenames.values():
-            self.assertTrue(fname in local_metadata)
-
-    def test_keys_and_signatures(self) -> None:
```

Comment:
There are some tests covering keys and signatures (like `test_new_timestamp_fast_foward_recovery`, `test_root_rotation`, etc) , but I'm not sure they follow the exact specific case covered here. I'm fine with deleting it though, if others find it unnecessary.

Reply:
I think this is fine: this specific test was just me adding an example of how to modify sigs and keys when using the simulator. I think the simulator documentation and the better tests that now exist cover all that better by now.